packages/core/components/AutoField/fields/SelectField/index.tsx

```diff
@@ -27,18 +28,20 @@ export const SelectField = ({
     >
       <select
         id={id}
+        title={label || name}
         className={getClassName("input")}
         disabled={readOnly}
-        onChange={(e) => {
+        onChange={({ target: { value } }) => {
           if (
-            e.currentTarget.value === "true" ||
-            e.currentTarget.value === "false"
+          value === "true" ||
+          value === "false" ||
+          typeof safeJsonParse(value) === "number"
```

Comment:
This is smart. I suppose we could reuse this for the boolean.

In fact, we could consider _always_ returning JSON if it's valid 🤔 that would add support for objects, too.

Reply:
Agreed, that is my preference I would actually prefer always returning if valid and if not returning the value as is, and not doing any typeof check.